cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -592,6 +593,7 @@ class alignas(16) mutable_column_device_view : public detail::column_device_view
     return d_children[child_index];
   }
 
+#ifdef __CUDACC__  // because set_bit in bit.hpp is wrapped with __CUDACC__
```

Comment:
These are necessary even though this is a `.cuh` file?

I thought `.cuh` files could only be included by `.cu` files, compiled by `nvcc`, so `__CUDACC__` would always be defined?

Reply:
My understanding is that is the guideline as well. However, this is overwhelming not followed (which I discovered from workin on this PR). For instance, `column_wrapper.hpp` where the `make_counting_transform_iterator` function used to be, has a .hpp, and the majority of the test files that used it were also .hpp.

Reply:
So you're saying `column_wrapper.hpp` and other `.hpp` files are doing `#include <cudf/detail/iterator.cuh>`? That seems like a problem.

Reply:
I agree, but sort of out of scope of this PR. Although ... I just did 
```
cat $(find . | grep "p$") | grep thrust | wc -l
```
in both `cpp/src` and `cpp/include` and the total was 53. However, upon brief review ... most of the references were to `thrust::host_vector` or thrust::make_**_iterator` so maybe those aren't consider "device" code?
___
Just did the same `grep` on `cpp/tests` and got 533 :open_mouth:  
```
cat $(find . | grep "p$") | grep __device__ | wc -l
```
yields 27 across `src` `include` and `tests`.

Reply:
@trxcllnt I will set up a issue / PR to rename filenames / look into this.